esp/services/esdl_svc_engine/esdl_binding.cpp

```diff
@@ -1454,18 +1529,22 @@ void EsdlServiceImpl::prepareFinalRequest(IEspContext &context,
     if (m_serviceLevelCrtFail)
         throw MakeStringException(-1, "%s::%s disabled due to service-level Custom Transform errors. Review transform template in configuration.", srvdef.queryName(), mthName);
 
-    StringAttr *crtErrorMessage = m_methodCRTransformErrors.getValue(mthName);
+    StringAttr *crtErrorMessage = m_requestTransformErrors.getValue(mthName);
     if (crtErrorMessage && !crtErrorMessage->isEmpty())
         throw MakeStringException(-1, "%s::%s disabled due to method-level Custom Transform errors: %s. Review transform template in configuration.", srvdef.queryName(), mthName, crtErrorMessage->str());
 
     if (scriptContext)
     {
-        IEsdlCustomTransform *methodCrt = m_customRequestTransformMap.getValue(mthName);
+        IEsdlTransformSet *serviceCRTs = m_transforms->queryMethodEntryPoint("", ESDLScriptEntryPoint_Legacy);
+        IEsdlTransformSet *methodCRTs = m_transforms->queryMethodEntryPoint(mthName, ESDLScriptEntryPoint_Legacy);
+
+        IEsdlTransformSet *serviceRTs = m_transforms->queryMethodEntryPoint("", ESDLScriptEntryPoint_BackendRequest);
+        IEsdlTransformSet *methodRTs = m_transforms->queryMethodEntryPoint(mthName, ESDLScriptEntryPoint_BackendRequest);
 
         context.addTraceSummaryTimeStamp(LogNormal, "srt-custreqtrans");
         scriptContext->setContent(ESDLScriptCtxSection_ESDLRequest, reqProcessed.str());
-        if (m_serviceLevelRequestTransform || methodCrt) //slightly redundant, but less fragile to future changes to check again here
-            processServiceAndMethodTransforms(scriptContext, {m_serviceLevelRequestTransform, methodCrt}, ESDLScriptCtxSection_ESDLRequest, ESDLScriptCtxSection_FinalRequest);
+        if (serviceCRTs || serviceRTs || methodCRTs || methodRTs)
+            processServiceAndMethodTransforms(scriptContext, {serviceCRTs, serviceRTs, methodCRTs, methodRTs}, ESDLScriptCtxSection_ESDLRequest, ESDLScriptCtxSection_FinalRequest);
         scriptContext->toXML(reqProcessed.clear(), ESDLScriptCtxSection_FinalRequest);
```

Comment:
I think this should be part of the `if (serviceCRTs || serviceRTs || methodCRTs || methodRTs)` clause. Otherwise when there are no transforms (Legacy or BackendRequest) we will clear out the reqProcessed which is just about to be sent to the back end.

Reply:
agreed